src/components/CircleProgressBar/index.js

```diff
@@ -15,11 +15,26 @@ const CircleProgressBar = React.createClass({
     aria: {}
   }),
 
+  restart () {
+    this.spinPercentage += 0.5;
+
+    this.progressBar.animate(this.spinPercentage);
+  },
+
+  infiniteSpin () {
+    this.progressBar.animate(this.spinPercentage, null, this.restart);
+  },
+
   componentDidMount () {
     this.initProgressBar();
     if (this.props.percentage > 0) {
       this.updateProgressBar(this.props.percentage);
     }
+
+    if (this.props.isSpinner) {
+      this.spinPercentage = 0.25;
+      this.interval = setInterval(this.infiniteSpin, 1000);
```

Comment:
couldn’t you use RequestAnimationFrame here? With a much lower spinPercentage obviously.